homeassistant/components/recorder/util.py

```diff
@@ -497,6 +514,10 @@ def setup_connection_for_dialect(
                 _fail_unsupported_version(
                     version or version_string, "SQLite", MIN_VERSION_SQLITE
                 )
+            if not version or version < UPCOMING_MIN_VERSION_SQLITE:
```

Comment:
```suggestion
            elif version < UPCOMING_MIN_VERSION_SQLITE:
```

I would not add a warning if version is None

Reply:
addressed in daa106ddcc823379e291e3ed5ce70cc396b9abc0